### ISSUE TITLE
Consistent URI/body decoding

### ISSUE THREAD

**Issue by snicoll** [spring-projects/spring-boot#1182]
Spring MVC defaults to `ISO-8859-1` for historical reason and it's not easy to change that value. The decoding of the URI happens at two stages: the container does it (see #542 for what we did for Tomcat since Jetty already defaults to `UTF-8`) and Spring MVC does it, based on the request encoding or the default if it's not set.

What people usually do is configure the `org.springframework.web.filter.CharacterEncodingFilter` with the same encoding as the one defined on the container. That way the body and the URI are decoded in a consistent manner.

Long story short: #542 does not fully provide a `UTF-8` decoding by default. It would be nice to try to bring that property back at a generic level if we happen to be able to configure jetty: that way we can customize the uriEncoding with the use of a single property.
